scripts/pre_push_hook.sh

```diff
@@ -0,0 +1,125 @@
+# This scripts helps running pre-push hook inside docker container by,
+# working as middle layer beteen hook and pre_push_hook.py in docker
+# If you can't understand whats going on here, please visit
+# https://docs.google.com/document/d/1whHSUEchpmnqxSsRvHU_WYbBzeqVMh7vrjpuA8FMnAM/edit?usp=sharing
+#
+# To install hook (which runs inside docker container), run this script
+# command from oppia root directory with --install argument
+# 
+# Currently it only works for Unix systems, on windows it will create symlink but won't have any effect
+
+DEV_CONTAINER="dev-server"
+DOCKER_EXEC_COMMAND="docker compose exec -T $DEV_CONTAINER "
+
+# Hook file paths, we need you use relative path as actual path won't be same in host and container
+PYTHON_PREPUSH_HOOK_PATH=".git/hooks/pre-push-python"
+PREPUSH_HOOK_PATH=".git/hooks/pre-push"
+
+# Get the oppia app directory
+OPPIA_DIR=$(pwd)
+
+# Location of git hooks directory
+HOOKS_DIR="$OPPIA_DIR/.git/hooks"
+
+# Path for pre-push hook file
+PRE_PUSH_FILE="$HOOKS_DIR/pre-push"
+PYTHON_PRE_PUSH_FILE="$HOOKS_DIR/pre-push-python"
+
+# Install pre-push hook
+install_hook() {
+    # Create symlinks for pre-push hook files
+    for file in "$PRE_PUSH_FILE" "$PYTHON_PRE_PUSH_FILE"; do
+        # Check if pre-push file is already a symlink
+        if [ -h "$file" ] && [ -e "$file" ]; then
+            echo "Symlink already exists (for $file)"
+
+        else
+            # If broken symlink, remove it
+            if [ -h "$file" ] && [ ! -e "$file" ]; then
+                rm "$file"
+                echo "Removed broken symlink (for $file)"
+            fi
+
+            # Try creating a symlink
+            base_symlink="../../scripts"  # Symlinks needs to be relative as actual path won't be same on host and docker container
+            if [ "$(basename $file)" == "pre-push" ]; then
+                ORIGINAL_FILE="$base_symlink/pre_push_hook.sh"  
+            else
+                ORIGINAL_FILE="$base_symlink/pre_push_hook.py"
+            fi
+
+                     ln -s "$ORIGINAL_FILE" "$file" &&
+                echo "Created symlink in .git/hooks directory" ||
+                {
+                    # Fallback to copy on windows
+                    cp "$ORIGINAL_FILE" "$file"
+                    echo "Copied file to .git/hooks directory"
+                }
+
+            # Make the hook file executable
+            chmod +x "$file"
+
+            if [ $? -eq 0 ]; then
+                echo "pre-push hook file is now executable!"
+            else
+                echo >&2 "Failed to make pre-push executable"
+                exit 1
+            fi
+        fi
+    done
+    exit 0
+}
+
+# Check for --install in args and install pre-push hook if itC's found
+for arg in "$@"; do
+    if [ "$arg" == "--install" ]; then
+        install_hook
+    fi
+done
+
+# Bypassing notification
+echo "You can Bypass Pre-Push check by using \`git push --no-verify\`"
```

Comment:
Please remove this. We expect changes to pass the tests to push the commits.

Reply:
done!